app/src/main/java/com/example/cosc345project/ui/screens/ProductScreen.kt

```diff
@@ -117,9 +123,29 @@ fun RetailerSlot(
                     DropdownMenuItem(
                         text = { Text(retailer.name!!) },
                         onClick = { },
-                        colors = MenuDefaults.itemColors(disabledTextColor = MaterialTheme.colorScheme.onSurface),
+                        //colors = MenuDefaults.itemColors(disabledTextColor = MaterialTheme.colorScheme.onSurface),
                         enabled = false
                     )
+
+                    if (store.latitude != null && store.longitude != null) {
+                        DropdownMenuItem(
+                            text = {
+                                Text("Show on Map")
+                            },
+                            onClick = {
+                                val geoLocation =
+                                    Uri.parse("geo:0,0?q=" + store.latitude!! + "," + store.longitude!! + "(" + retailer.name!! + ")&z=12")
+
+                                val intent = Intent(Intent.ACTION_VIEW).apply {
+                                    data = geoLocation
+                                }
+
+    //                            if (intent.resolveActivity(context.packageManager) != null) {
```

Comment:
I think we do need to check if there's a map app, as otherwise I think there would be a crash